### PR TITLE
Fix a few issues which can lead to undefined-behaviour

### PR DESCRIPTION
**Pull request by bzsolt**
JerryScript-DCO-1.0-Signed-off-by: Zsolt Borbély zsborbely.u-szeged@partner.samsung.com

### CODE REVIEW

File: jerry-core/parser/regexp/re-bytecode.c [hunk 1]
```diff
@@ -126,7 +126,8 @@ re_bytecode_list_insert (re_bytecode_ctx_t *bc_ctx_p, /**< RegExp bytecode conte
 inline ecma_char_t __attr_always_inline___
 re_get_char (uint8_t **bc_p) /**< pointer to bytecode start */
 {
-  ecma_char_t chr = *((ecma_char_t *) *bc_p);
+  ecma_char_t chr;
+  memcpy (&chr, *bc_p, sizeof (ecma_char_t));
```
Comment (akosthekiss) [inline jerry-core/parser/regexp/re-bytecode.c]: What exactly was undefined behaviour in the original code? Is memcpy the only way to get around it? As it is a call to a libc function, it is much more expensive than a simple cast.

Comment (zherczeg) [inline jerry-core/parser/regexp/re-bytecode.c]: Reading uint16 from byte aligned address is undefined in C.

Comment (akosthekiss) [inline jerry-core/parser/regexp/re-bytecode.c]: Ah, that. But cannot we have a macro that does the uint16 read from a byte pointer "in place" without the function call overhead (and the loop overhead of memcpy)?

Comment (akosthekiss) [inline jerry-core/parser/regexp/re-bytecode.c]: Any answer to this, even though it got merged already?...

Comment (zherczeg) [inline jerry-core/parser/regexp/re-bytecode.c]: As far as I know most compilers are proud to optimize this.

Comment (zherczeg) [inline jerry-core/parser/regexp/re-bytecode.c]: If we don't trust them we could create a macro. Problem is, if the architecture supports unaligned read, we are expecting the compiler to deoptimize our macro.

Comment (zherczeg) [inline jerry-core/parser/regexp/re-bytecode.c]: Btw GCC has a flag to not optimize this: -fno-builtin-memcpy

Comment (akosthekiss) [inline jerry-core/parser/regexp/re-bytecode.c]: The thing is that we have our own memcpy implementation in jerry-libc and I'm not sure that gcc can optimize it in that case. Especially as we are compiling jerry with -fno-builtin, which has effect on memcpy as well.

**Pull request merged**